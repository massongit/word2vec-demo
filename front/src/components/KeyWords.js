import React from "react"
import {FormattedMessage} from "react-intl"
import PropTypes from "prop-types"

/**
 * positiveとnegativeのPropTypes
 * @type {{positive: shim, negative: shim}}
 */
export const propTypesPN = {
    positive: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    negative: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
}

/**
 * キーワード
 */
class KeyWords extends React.Component {
    static propTypes = propTypesPN

    render() {
        return (
            <p>
                <label htmlFor="keywords">
                    <FormattedMessage id="header.keywords"/>
                </label>
                <p id="keywords">
                    {
                        (() => {
                            let r = []

                            if (this.props.positive) {
                                r = r.concat(this.props.positive.map(w => (
                                    "+" + w
                                )))
                            }

                            if (this.props.negative) {
                                r = r.concat(this.props.negative.map(w => (
                                    "-" + w
                                )))
                            }

                            return r.join(" ")
                        })()
                    }
                </p>
            </p>
        )
    }
}

export default KeyWords
