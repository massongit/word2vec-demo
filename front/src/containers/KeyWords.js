import KeyWords from "../components/KeyWords"
import {connect} from "react-redux"

const mapStateToProps = state => ({
    positive: state.showSimilarWords.words.positive,
    negative: state.showSimilarWords.words.negative
})

/**
 * キーワードのContainer
 */
export default connect(mapStateToProps)(KeyWords)
