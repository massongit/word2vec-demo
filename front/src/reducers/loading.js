import * as types from "../actions/types"
import {handleActions} from "redux-actions"

/**
 * ローディングActionのReducer
 */
export default handleActions(
    {
        // ローディングActionのとき
        [types.LOADING]: (state, action) => {
            if (typeof action.payload.loading === "boolean") {
                return {
                    loading: action.payload.loading
                }
            } else {
                return state
            }
        }
    },
    {
        loading: false
    }
)
