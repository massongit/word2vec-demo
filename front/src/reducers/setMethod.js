import * as types from "../actions/types"
import {handleActions} from "redux-actions"

/**
 * 手法セットActionのReducer
 */
export default handleActions(
    {
        // 手法セットActionのとき
        [types.SET_METHOD]: (state, action) => {
            if (action.payload.method) {
                return {
                    prevMethod: state.method,
                    method: action.payload.method
                }
            } else {
                return state
            }
        }
    }, {}
)
