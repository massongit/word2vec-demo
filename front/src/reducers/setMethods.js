import * as types from "../actions/types"
import {handleActions} from "redux-actions"

/**
 * 手法一覧セットActionのReducer
 */
export default handleActions(
    {
        // 手法一覧セットActionのとき
        [types.SET_METHODS]: (state, action) => {
            if (0 < action.payload.methods.length) {
                return {
                    methods: action.payload.methods
                }
            } else {
                return state
            }
        }
    },
    {
        methods: []
    }
)
