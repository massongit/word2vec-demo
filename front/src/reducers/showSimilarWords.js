import * as types from "../actions/types"
import {handleActions} from "redux-actions"

/**
 * Word2Vecによる計算結果の表示ActionのReducer
 */
export default handleActions(
    {
        // Word2Vecによる計算結果の表示Actionのとき
        [types.SHOW_SIMILAR_WORDS]: (state, action) => {
            if (action.payload.words && action.payload.words.similar) {
                for (const s of action.payload.words.similar) {
                    if (!s || !s.word || !s.cosine) {
                        return state
                    }
                }

                if (action.payload.keywords && action.payload.words && !action.payload.words.similar.includes(undefined)) {
                    return {
                        keywords: action.payload.keywords,
                        words: action.payload.words
                    }
                }
            }

            return state
        }
    },
    {
        keywords: {
            positive: [],
            negative: []
        },
        words: {
            positive: [],
            negative: [],
            similar: []
        }
    }
)
