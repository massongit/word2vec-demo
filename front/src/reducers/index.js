import showSimilarWords from "./showSimilarWords"
import {combineReducers} from "redux"

/**
 * ルートReducer
 */
export default combineReducers({
    showSimilarWords
})
