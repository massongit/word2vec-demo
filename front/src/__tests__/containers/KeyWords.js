import React from "react"
import KeyWords from "../../containers/KeyWords"
import rootReducer from "../../reducers"
import {shallow} from "enzyme"
import {createStore} from "redux"
import {showSimilarWordsState} from "../../test_data"
import {dispatchActions, makeShowSimilarWordsAction} from "../reducers"

export const functions = store => {
    dispatchActions(store, makeShowSimilarWordsAction(showSimilarWordsState))
}

export const doSnapshot = (fs, store, g) => {
    for (const f of fs) {
        if (f) {
            f(store)
        }

        expect(g()).toMatchSnapshot()
    }
}

let store

/**
 * テストの前処理
 */
const beforeProcess = () => (
    shallow(
        <KeyWords
            store={store}
        />
    ).dive()
)

describe("containers/KeyWords", () => {
    beforeEach(() => {
        store = createStore(rootReducer)
    })


    it("初期状態からStateが遷移した際に、Componentが正しく配置されている", () => {
        doSnapshot([functions], store, beforeProcess)
    })
})
