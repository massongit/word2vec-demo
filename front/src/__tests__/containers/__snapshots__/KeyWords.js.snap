// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/KeyWords 初期状態からStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyWords
    dispatch={[Function]}
    negative={
        Array [
            "男",
          ]
    }
    positive={
        Array [
            "王",
            "女",
          ]
    }
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="keywords"
>
          <FormattedMessage
                    id="header.keywords"
                    values={Object {}}
          />
</label>,
        <p
          id="keywords"
>
          +王 +女 -男
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FormattedMessage
            id="header.keywords"
            values={Object {}}
/>,
          "htmlFor": "keywords",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "header.keywords",
            "values": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+王 +女 -男",
          "id": "keywords",
        },
        "ref": null,
        "rendered": "+王 +女 -男",
        "type": "p",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="keywords"
>
            <FormattedMessage
                        id="header.keywords"
                        values={Object {}}
            />
</label>,
          <p
            id="keywords"
>
            +王 +女 -男
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FormattedMessage
              id="header.keywords"
              values={Object {}}
/>,
            "htmlFor": "keywords",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "header.keywords",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+王 +女 -男",
            "id": "keywords",
          },
          "ref": null,
          "rendered": "+王 +女 -男",
          "type": "p",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
